opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/service/xs2a/consent/Xs2aConsentInfo.java

```diff
@@ -41,4 +41,13 @@ public boolean isWrongScaChallenge(Xs2aContext ctx) {
     public boolean isOkRedirectConsent(Xs2aContext ctx) {
         return ctx.isRedirectConsentOk();
     }
+
+    public boolean isOauthAuthorizationCodePresent(Xs2aContext ctx) {
```

Comment:
Not sure if this applies to your case, but for Oauth2 case class StartConsentAuthorization, section
```
Response<StartScaProcessResponse> scaStart = ais.startConsentAuthorisation(
                params.getPath().getConsentId(),
                params.getHeaders().toHeaders()
        );
```

scaStart should contain `scaOAuth` header if Oauth2 was chosen, according to:

![Selection_092](https://user-images.githubusercontent.com/18463491/87848065-62ddf200-c8e5-11ea-880d-1f24901a3dc5.png)

If the header is present at that step, then I suggest to:
1. Add `scaOauthLink` field to `Xs2aContext`
2. at StartConsentAuthorization, do:
```
context.setScaOauthLink(scaStart.getBody().getLinks().get("scaOauth").getHref());
```
3. For isOauthAuthorizationCodePresent function:
```
public boolean isOauthApproach(Xs2aContext ctx) {
   return !Strings.isNullOrEmpty(ctx.getScaOauthLink());
}
```

And to have the following process for authorization:
![Selection_093](https://user-images.githubusercontent.com/18463491/87848311-a20d4280-c8e7-11ea-97cd-e87b17ee4bfe.png)